dotcom-rendering/src/components/Timeline.tsx

```diff
@@ -0,0 +1,259 @@
+	width: 12px;
+	height: 12px;
+	content: '';
+	border: 1px solid ${palette('--timeline-event-border')};
+	position: absolute;
+	left: -7px;
+	top: -6px;
+`;
+
+const smallDateStyles = css`
+	display: block;
+	${textSans.small({ fontWeight: 'bold' })}
+
+	${from.desktop} {
+		${textSans.medium({ fontWeight: 'bold' })}
+	}
+`;
+
+const titleWeight = ({ design }: ArticleFormat): 'bold' | 'medium' => {
+	switch (design) {
+		case ArticleDesign.Feature:
+		case ArticleDesign.Interview:
+		case ArticleDesign.Recipe:
+		case ArticleDesign.Review:
+			return 'bold';
+		default:
+			return 'medium';
+	}
+};
+
+const titleStyles = (format: ArticleFormat): SerializedStyles => css`
+	${headline.xxsmall({ fontWeight: titleWeight(format) })}
+
+	${from.desktop} {
+		${headline.xsmall({ fontWeight: titleWeight(format) })}
+	}
+`;
+
+type EventHeaderProps = {
+	event: DCRTimelineEvent;
+	ArticleElementComponent: NestedArticleElement;
+	sectioned: boolean;
+	smallDates: boolean;
+	format: ArticleFormat;
+};
+
+const EventHeader = ({
+	event,
+	ArticleElementComponent,
+	format,
+	sectioned,
+	smallDates,
+}: EventHeaderProps) => {
+	const heading = (
+		<Heading level={sectioned ? 3 : 2}>
+			<span css={smallDates ? smallDateStyles : titleStyles(format)}>
+				{event.date}
+			</span>
+			{event.title !== undefined ? (
+				<span css={titleStyles(format)}>{event.title}</span>
+			) : null}
+		</Heading>
+	);
+
+	if (event.main !== undefined && hasShowcaseRole(event.main)) {
+		return (
+			<header>
+				<ArticleElementComponent
+					index={0}
+					element={event.main}
+					format={format}
+				/>
+				{heading}
+			</header>
+		);
+	} else if (event.main !== undefined) {
+		return (
+			<header>
+				{heading}
+				<ArticleElementComponent
+					index={0}
+					element={event.main}
+					format={format}
+				/>
+			</header>
+		);
+	} else {
+		return heading;
+	}
+};
+
+// ----- TimelineEvent ----- //
+
+const eventStyles = css`
+	border: 1px solid ${palette('--timeline-event-border')};
+	padding: ${space[1]}px 10px ${space[6]}px 10px;
+	margin-bottom: ${space[5]}px;
+	position: relative;
+
+	&:before {
+		${timelineBulletStyles}
+	}
+`;
+
+const labelStyles = css`
+	border: 1px solid ${palette('--timeline-event-border')};
+	border-bottom: none;
+	padding: 3px 10px 4px 10px;
+	display: inline-block;
+	${textSans.small({ fontWeight: 'regular' })}
+`;
+
+const immersiveMainElementEventStyles = css`
+	padding-top: 0;
+`;
+
+type TimelineEventProps = {
+	event: DCRTimelineEvent;
+	ArticleElementComponent: NestedArticleElement;
+	sectioned: boolean;
+	smallDates: boolean;
+	format: ArticleFormat;
+};
+
+const TimelineEvent = ({
+	event,
+	ArticleElementComponent,
+	sectioned,
+	smallDates,
+	format,
+}: TimelineEventProps) => (
+	<>
+		{event.label !== undefined && (
+			<div css={labelStyles}>{event.label}</div>
+		)}
+		<section
+			css={[
+				eventStyles,
+				hasShowcaseRole(event.main)
+					? immersiveMainElementEventStyles
+					: undefined,
```

Comment:
Emotion will strip out falsey values if you prefer:

```suggestion
				hasShowcaseRole(event.main)
					&& immersiveMainElementEventStyles
```

Reply:
I'm afraid I'm not keen on using implicit conversions and `boolean` operators for values that aren't booleans. I don't find the code very clear, and it can sometimes cause subtle bugs.